code/script.js

```diff
@@ -251,93 +274,168 @@ const setSecret = () => {
-const filterCharacters = (keep) => {
-  // Show the correct alert message for different categories
-  if (group === 'accessories') {
+
+const filterCharacters = (keep) => {    // Will filter the characters array and redraw the game board.
+
+
+  if (currentQuestion.category === 'accessories') {     // Shows the correct alert message for different categories
     if (keep) {
       alert(
-        `Yes, the person wears ${attribute}! Keep all that wears ${attribute}`
+        `Yes, the person wears ${currentQuestion.label}! I'll remove all who don't for you.`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${attribute}! Remove all that wears ${attribute}`
+        `No, the person doesn't wear ${currentQuestion.label}! I'll remove all who do for you.`
+      )
+    }
+  } else if (currentQuestion.category === 'other') {
+    if (keep) {
+      alert(
+        `Yes, the person ${currentQuestion.label}s! I'll remove all who don't for you.`
+      )
+    } else {
+      alert(
+        `No, the person doesn't ${currentQuestion.label}! I'll remove all who do for you.`
       )
     }
-  } else if (group === 'other') {
-    // Similar to the one above
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all persons with yellow hair"
+      alert(
+        `Yes, the person has ${currentQuestion.value} ${currentQuestion.label}! I'll remove everyone who don't for you.`
+      )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(
+        `Nope, the person doesn't have ${currentQuestion.value} ${currentQuestion.label}! I'll remove everyone who does for you.`
+      )
     }
+
   }
 
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+  //// The actual filtering ////
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value) 
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value)
+  }
+
+  generateBoard();
 
-  // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  
+  const makeGuess = confirm(`Are you sure you want to guess on ${suspect} being the secret person?`)
+  
+  if (makeGuess) {
+
+    checkMyGuess(suspect)
+
+  } else {
+    //intentionally kept empty
+  }
```

Comment:
I don't think we need to add an else if we are not going to use it. However, good too keep it if you want to add something to it later on.